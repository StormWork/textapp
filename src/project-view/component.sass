.project-view
  display: grid
  grid-template-columns: 1fr 8pt 1fr
  grid-gap: 8pt
  margin: 11px

  &-grid
    display: grid
    grid-template-columns: 7fr 3fr 3fr
    grid-gap: 16px

    &__column-1
      grid-column: 1

    &__column-1-2
      grid-column: 1 / 3

    &__column-2
      grid-column: 2

    &__column-3
      grid-column: 3
      display: flex
      flex-direction: column
      align-items: center

  &__button
    background-color: #6441A4
    border: 0
    color: white
    height: 15pt
    min-width: 8em

    &--next
      margin-left: 11px

  &__title
    font-size: larger
    font-weight: bold

  &-property
    display: block

    &__input
      margin-top: 4px
      width: 100%

    &__name
      display: block

    &__result
      margin: 2px 8px
      text-align: center

    &__value
      color: grey
      user-select: text

  &__vertical-bar
    background-color: #DAD8DE
    width: 1pt;
    height: 100%

  &__item
    margin-bottom: 8pt
    margin-left: -17pt

  &__item-text
    margin: 3pt

    &--title
      font-weight: bold
