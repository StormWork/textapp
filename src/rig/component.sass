.rig-container
  display: flex
  flex-direction: column
  height: 100%

label
  user-select: none
