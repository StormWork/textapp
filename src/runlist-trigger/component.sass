.runlist-trigger
  align-self: flex-end
  padding-right: 0.8em

  &__button
    color: white
    background-color: #7D55C7
    border: 1px solid #7D55C7
    text-align: center
    width: 4rem
    height: 2rem
    vertical-align: middle
    cursor: pointer
    font-size: .75em

  &__text
    display: inline
    color: #7D55C7
    font-size: smaller
    opacity: 1
    visibility: hidden

    &--on
      visibility: visible

    &--fading
      opacity: 0
      transition: 2s

  &__select
    color: #7D55C7
    background: transparent
    border: none
    font-size: 14px
    height: 29px
    padding: 5px
    margin-right: 1rem
    text-align-last: right
