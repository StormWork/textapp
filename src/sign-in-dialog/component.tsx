import * as React from 'react';
import './component.sass';
import { LoginButton } from '../login-button';

export class SignInDialog extends React.Component {
  public render() {
    return (
      <div className="sign-in">
        <div className="sign-in__background" />
        <div className="sign-in__dialog">
          <div className="dialog__top-bar-container">
            <div className="top-bar-container__title"> Please Sign In </div>
          </div>
          <div>
            <hr className="dialog__divider" />
            <div className='sign-in__text'>
              <p>Why do I need to sign in?</p>
              <p>We built the Developer Rig with one goal in mind: helping developers build extensions faster.  We want to make sure you’re spending your time coding, not on process.  To take advantage of the Rig’s features, we need you to sign in.  You can look forward to more new features the coming months.</p>
              <p>We really appreciate you being part of the Twitch developer community. We’re excited to see what kind of extensions you’ll make!</p>
            </div>
            <div className="sign-in__bottom-bar">
              <LoginButton />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
