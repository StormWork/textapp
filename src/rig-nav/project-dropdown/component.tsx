import * as React from 'react';
import * as classNames from 'classnames';
import * as whitetriangle from '../../img/whitetriangle.svg';
import './component.sass';
import { RigProject } from '../../core/models/rig';

interface State {
  open: boolean;
}

export interface Props {
  currentProjectIndex: number;
  projects: RigProject[];
  createNewProject: () => void;
  selectProject: (projectIndex: number) => void;
}

export class ProjectDropdown extends React.Component<Props, State> {
  state: State = {
    open: false,
  };

  private handleClick = () => {
    this.setState({ open: !this.state.open });
  }

  public render() {
    const dropdownClass = classNames({
      'project__menu': true,
      'transition': true,
      'open': this.state.open,
    });

    const triangleClass = classNames({
      'project__triangle': true,
      'open': this.state.open,
    });

    return (
      <div className='project__dropdown' onClick={this.handleClick}>
        <div className='project__header'>Select an Extension Project <img className={triangleClass} src={whitetriangle} alt='open or close' /></div>
        <div className={dropdownClass}>
          <ul>
            <li key={-1}>
              <div onClick={this.props.createNewProject}>Create New Project</div>
            </li>
            {this.props.projects.map((project, index) => (
              <li key={index}>
                <div title={project.manifest.description} onClick={() => this.props.selectProject(index)}>{project.manifest.name}</div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    );
  }
}
