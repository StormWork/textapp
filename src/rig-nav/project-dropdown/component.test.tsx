import { setupShallowTest } from '../../tests/enzyme-util/shallow';
import { ProjectDropdown } from './component';

describe('<ProjectDropdown />', () => {
  const setupShallow = setupShallowTest(ProjectDropdown, () => ({
    projects: [],
  }));

  it('renders correctly', () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });
});
