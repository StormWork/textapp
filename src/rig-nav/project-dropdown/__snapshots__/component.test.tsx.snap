// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectDropdown /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectDropdown
    projects={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="project__header"
>
          Select an Extension Project 
          <img
                    alt="open or close"
                    className="project__triangle"
                    src="whitetriangle.svg"
          />
</div>,
        <div
          className="project__menu transition"
>
          <ul>
                    <li>
                              <div
                                        onClick={undefined}
                              >
                                        Create New Project
                              </div>
                    </li>
          </ul>
</div>,
      ],
      "className": "project__dropdown",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Select an Extension Project ",
            <img
              alt="open or close"
              className="project__triangle"
              src="whitetriangle.svg"
/>,
          ],
          "className": "project__header",
        },
        "ref": null,
        "rendered": Array [
          "Select an Extension Project ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "open or close",
              "className": "project__triangle",
              "src": "whitetriangle.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
                        <div
                                    onClick={undefined}
                        >
                                    Create New Project
                        </div>
            </li>
</ul>,
          "className": "project__menu transition",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <div
                                onClick={undefined}
                >
                                Create New Project
                </div>
</li>,
              Array [],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "-1",
              "nodeType": "host",
              "props": Object {
                "children": <div
                  onClick={undefined}
>
                  Create New Project
</div>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create New Project",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Create New Project",
                "type": "div",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="project__header"
>
            Select an Extension Project 
            <img
                        alt="open or close"
                        className="project__triangle"
                        src="whitetriangle.svg"
            />
</div>,
          <div
            className="project__menu transition"
>
            <ul>
                        <li>
                                    <div
                                                onClick={undefined}
                                    >
                                                Create New Project
                                    </div>
                        </li>
            </ul>
</div>,
        ],
        "className": "project__dropdown",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Select an Extension Project ",
              <img
                alt="open or close"
                className="project__triangle"
                src="whitetriangle.svg"
/>,
            ],
            "className": "project__header",
          },
          "ref": null,
          "rendered": Array [
            "Select an Extension Project ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "open or close",
                "className": "project__triangle",
                "src": "whitetriangle.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                            <div
                                          onClick={undefined}
                            >
                                          Create New Project
                            </div>
              </li>
</ul>,
            "className": "project__menu transition",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <div
                                    onClick={undefined}
                  >
                                    Create New Project
                  </div>
</li>,
                Array [],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "-1",
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    onClick={undefined}
>
                    Create New Project
</div>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Create New Project",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Create New Project",
                  "type": "div",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
